.travis/osx..script.sh

```diff
@@ -5,4 +5,4 @@ if [ $QT5 ]; then
         export CMAKE_PREFIX_PATH="$(brew --prefix qt55)"
 fi
 
-cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWANT_QT5=$QT5 -DUSE_WERROR=OFF ..
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWANT_QT5=$QT5 -DUSE_WERROR=ON ..
```

Comment:
To be more specific, couldn't we default it for `DCMAKE_BUILD_TYPE=Debug` like we do for stripping binaries, or is this flag a useful one to have around regardless?